### ISSUE TITLE
job_configuration_location column is not defined in batch_job_execution table in all database migration scripts

### ISSUE THREAD

**Issue by mabrod** [spring-projects/spring-batch#4042]
**Bug description**
Missing column job_configuration_location prevents insert into batch_job_execution table when running a batch job with one of the databases as job repository.

**Environment**
All database scripts on the main branch including schema-postgresql.sql which I use for my job repository when running batch jobs do not have job_configuration_location column defined in batch_job_execution table.

Once I have added this column to the table with this SQL statement everything works fine.
ALTER TABLE batch_job_execution
ADD COLUMN job_configuration_location VARCHAR(2500) NULL;
I had used alter table cause I had already run the scripts to create necessary job repository tables in Postgres database, but 
the column should be added to create table definition statement.

**Comment by cppwfs** [comment]
Can you share the version of Batch you are using?   The column is present in the postgresql schema for the 4.3.4 release of Spring Batch .
https://github.com/spring-projects/spring-batch/blob/4.3.4/spring-batch-core/src/main/resources/org/springframework/batch/core/schema-postgresql.sql#L22

**Comment by mabrod** [comment]
At that point I had used spring boot 2.6.2 which pulled in spring batch core 4.3.4 version and you are right that version contains JOB_CONFIGURATION_LOCATION column. Everything is fine when I use postgres scripts to create batch tables from this spring batch version.
I have used postgres schema scripts from main branch
https://github.com/spring-projects/spring-batch/tree/main/spring-batch-core/src/main/resources/org/springframework/batch/core
I think that was a problem to begin with cause the scripts in the main lack this column and I should've used scripts from the spring boot batch 4.3.4 version.
Cheers,
Matt
